railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -115,6 +117,17 @@ def extract_environment_from_path(path)
           available_environments.find { |env| path.include? env } if path.end_with?(".yml.enc")
         end
 
+        # Tries to check for mismatch between the provided `environment` class_option and the
+        # Rails::Command.environment. When no class_option for environment is provided, by default,
+        # it works with the development environment. So, ideally warning the user on their performing
+        # action could avoid unnecessary confusions. This warning happens only in the non-development
+        # environment.
+        def warn_if_environment_mismatch(method_name: :show)
+          if Rails::Command.environment != "development" && Rails::Command.environment != options[:environment]
+            say "You are accessing #{options[:environment]} credentials from the #{Rails::Command.environment} environment. Did you mean to run `rails credentials:#{method_name} --environment #{Rails::Command.environment}`? For more information try `rails credentials:help`"
```

Comment:
Knit: but could we line-break this string?

Something like `"you are accessing....." \ "blah blah" \ "foo bar"?

Reply:
@zzak Introduced line breaks to make it little more clear to the user. Thanks for the suggestion.